include/picongpu/fields/FieldJ.kernel

```diff
@@ -211,7 +211,7 @@ namespace picongpu
                 const DataSpace<simDim> localCell(DataSpaceOperations<simDim>::template map<TVec>(particleCellIdx));
 
                 Velocity velocity;
-                const float3_X vel = velocity(particle[momentum_], attribute::getMass(weighting, particle));
+                const float3_X vel = velocity(static_cast<float3_X>(particle[momentum_]), attribute::getMass(weighting, particle));
```

Comment:
Is this cast really necessary? As far as I know momentum_ already should be float3_X

Reply:
Yes, this is necessary when `particle[momentum_]` returns a [proxy reference](https://llama-doc.readthedocs.io/en/latest/pages/proxyreferences.html), which it can with certain LLAMA mappings.